### PR TITLE
MONGOID-4521 Ensure that QueryCache doesn't cache multi-batch results

### PR DESCRIPTION
**Pull request by saghm**

### CODE REVIEW

File: spec/mongoid/query_cache_spec.rb [hunk 1]
```diff
@@ -414,15 +414,15 @@
   context "when querying a very large collection" do
 
     before do
-      123.times { Band.create! }
```
Comment (simi) [inline spec/mongoid/query_cache_spec.rb]: What's the reason for this change?

File: spec/mongoid/query_cache_spec.rb [hunk 2]
```diff
@@ -450,6 +450,19 @@
       Band.collection.indexes.create_one(name: 1)
     end
   end
+
+  context 'when the initial query does not exhaust the results' do
+    before do
+      Mongoid::QueryCache.enabled = true
```
Comment (simi) [inline spec/mongoid/query_cache_spec.rb]: Usually it is good idea to temporarily enable config option and restore original state once test is done (for example using `around` block).

https://github.com/mongodb/mongoid/blob/b3a5918e504f4ac9ae4e750701bb02db78000960/spec/mongoid/query_cache_spec.rb#L308-L313

Comment (saghm) [reply spec/mongoid/query_cache_spec.rb]: Note that at the top of the description block, there is already an `around` block defined:

```ruby
around do |spec|
    Mongoid::QueryCache.clear_cache
    Mongoid::QueryCache.cache { spec.run }
  end
```

The implementation of `QueryCache.cache` already saves the value from before the passed block and restores it after.

File: spec/mongoid/query_cache_spec.rb [hunk 3]
```diff
@@ -452,6 +452,19 @@
...
+      10.times { Band.create! }
+
+      Band.batch_size(4).all.any?
+    end
+
+    it 'does not cache the result' do
+      expect(Band.all.map(&:id).size).to eq(10)
+    end
```
Comment (p-mongo) [inline spec/mongoid/query_cache_spec.rb]: This test is good but there should be another one setting batch size to 4 at query time.

File: spec/mongoid/query_cache_spec.rb (hunk 2 above)
Comment (p-mongo) [reply spec/mongoid/query_cache_spec.rb]: Is enabling query cache again on this line redundant then?

File: spec/mongoid/query_cache_spec.rb (hunk 3 above)
Comment (saghm) [reply spec/mongoid/query_cache_spec.rb]: I'm not sure I understand what that has to do with the query cache; is the idea to do a query with the same batch size as the number of results of the previous query?

Comment (saghm) [reply spec/mongoid/query_cache_spec.rb]: Done, as discussed offline
